src/snowflake/cli/plugins/nativeapp/lsp_commands.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import inspect
+from typing import Callable, TypedDict
+
+from pygls.server import LanguageServer
+from snowflake.cli.api.output.types import MessageResult
+from snowflake.cli.api.project.definition_manager import DefinitionManager
+from snowflake.cli.app.snow_connector import connect_to_snowflake
+from snowflake.cli.plugins.nativeapp.manager import NativeAppManager
+from snowflake.connector import SnowflakeConnection
+
+
+def lsp_plugin(name: str, version: str, capabilities: dict) -> Callable:
+    def _decorator(func: Callable) -> Callable:
+        setattr(
+            func,
+            "_lsp_context",
+            {
+                "lsp_plugin_name": name,
+                "lsp_plugin_version": version,
+                "lsp_plugin_capabilities": capabilities,
+            },
+        )
+        return func
+
+    return _decorator
+
+
+class ConnectionParams(TypedDict):
+    session_token: str
+    master_token: str
+    account: str
+    connection_name: str
+    params: dict
+
+
+def server_command(server: LanguageServer, command_name: str):
```

Comment:
Should this be in `plugins/lsp` rather than `plugins/nativeapp`?

Reply:
Yes! I moved the decorator and the parts intended to be used by other plugin areas to make their own lsp_plugins into the lsp/utils.py file. I'm not quite sure utils is the best name for it since other util files in this repo tend to be pretty small non-core utility functions, but the file has a collection of functions for use.